### ISSUE TITLE
[SOLVED] socket.emit inside ss(socket).emit(...) callback doesn't arrive

### ISSUE THREAD

**Issue by borsje** [nkzawa/socket.io-stream#59]
My Node.js application is working fine, and the piece using socket.io-stream was working fine untill I picked up development (since ~4 months inactivity).

First of all, I the browser uploads a file, the server receives it, reads it, and should notify the browser with some information. I can see the server receiving and reading the file(s) without any problem through the console. However, the browser doesn't receive the emits sent by the server. My code is as follows:
```javascript
ss(socket).on("purchase_image", function(stream, data) {
		console.log("-- image");
		...
		temp.open({
			suffix: suffix
		}, function(error, info) {
			if (!error) {
				stream.on("error", function(error) {
					console.log("error");
					console.log(error);
				});
				stream.on("end", function() {
					console.log("purchase_reading emitted");
					socket.emit("purchase_reading", {
						status: "start",
						tmp_name: info.path
					});
					...
```

Has something changed in the way I should be emitting data back to the client from within the socket.io-stream callback?

**Comment by nkzawa** [comment]
Do you mean you can't receive the `purchase_reading` event on browser? How your client code looks like?

**Comment by borsje** [comment]
Exactly, and my code is:
```javascript
socket.on("purchase_reading", function(data) {
	console.log(data);
	if (data.status == "start") {
		$("#progress_upload").data("status", 1).switchClass("", "progress-bar-success", function() {
			//For some reason the 100% width gets removed, so add it again afterwards
			$(this).css({width: "100%"})
		});
		
		$("#reading").data("status", 0).switchClass("label-default", "label-primary");
		$("#reading").find(".fa").addClass("fa-spin");
		$("[name=file_tmp_name]").val(data.tmp_name);
	} else {
		console.log("end");
		if ($("#progress_upload").data("status") == 0) {
			//The user has uploaded another document in the meanwhile
			return true;
		}
		$("#reading").data("status", 1).switchClass("label-primary", "label-success");
		$("#reading").find(".fa").removeClass("fa-spin");
	}
});
```
I've also tried wrapping 'socket' in ss(), but did not help either.

EDIT DURING WRITING: I just realised this event binder is not inside $(document).ready(...). As this piece of code is used on multiple places within the project, I "couldn't" wrap it inside ready(). I tried to confirm this, and the event arrives. I guess this has to do with some performance updates resulting in a "too quick" page load.
In other words, it has been solved, but thanks for your quick reply! If the issue won't help anyone, feel free to delete it ;).
